tests/phpunit/tests/html/wpHtmlTagProcessorBookmarks.php

```diff
@@ -0,0 +1,370 @@
+<?php
+/**
+ * Unit tests covering WP_HTML_Tag_Processor bookmark functionality.
+ *
+ * @package WordPress
+ * @subpackage HTML
+ */
+
+require_once ABSPATH . WPINC . '/wp-html.php';
+
+/**
+ * @group html
+ *
+ * @coversDefaultClass WP_HTML_Tag_Processor
+ */
+class WP_HTML_Tag_Processor_Bookmark_Test extends WP_UnitTestCase {
+
+	/**
+	 * @ticket 56299
+	 *
+	 * @covers set_bookmark
+	 */
+	public function test_set_bookmark() {
+		$p = new WP_HTML_Tag_Processor( '<ul><li>One</li><li>Two</li><li>Three</li></ul>' );
+		$p->next_tag( 'li' );
+		$this->assertTrue( $p->set_bookmark( 'first li' ), 'Could not allocate a "first li" bookmark.' );
+		$p->next_tag( 'li' );
+		$this->assertTrue( $p->set_bookmark( 'second li' ), 'Could not allocate a "second li" bookmark.' );
+		$this->assertTrue( $p->set_bookmark( 'first li' ), 'Could not move the "first li" bookmark.' );
+	}
+
+	/**
+	 * @ticket 56299
+	 *
+	 * @covers release_bookmark
+	 */
+	public function test_release_bookmark() {
+		$p = new WP_HTML_Tag_Processor( '<ul><li>One</li><li>Two</li><li>Three</li></ul>' );
+		$p->next_tag( 'li' );
+		$this->assertFalse( $p->release_bookmark( 'first li' ), 'Released a non-existing bookmark.' );
+		$p->set_bookmark( 'first li' );
+		$this->assertTrue( $p->release_bookmark( 'first li' ), 'Could not release a bookmark.' );
+	}
+
+	/**
+	 * @ticket 56299
+	 *
+	 * @covers seek
+	 * @covers set_bookmark
+	 */
+	public function test_seek() {
+		$p = new WP_HTML_Tag_Processor( '<ul><li>One</li><li>Two</li><li>Three</li></ul>' );
+		$p->next_tag( 'li' );
+		$p->set_bookmark( 'first li' );
+
+		$p->next_tag( 'li' );
+		$p->set_attribute( 'foo-2', 'bar-2' );
+
+		$p->seek( 'first li' );
+		$p->set_attribute( 'foo-1', 'bar-1' );
+
+		$this->assertEquals(
```

Comment:
```suggestion
		$this->assertSame(
```
As this is not comparing objects or object arrays, `assertSame()` should be used.

:1234: Applies elsewhere in the PR.